test/extended/storage/driver_configuration.go

```diff
@@ -0,0 +1,326 @@
+				clusterCSIDriverOptions: map[string]int{
+					snapshotOptions["vvol"]["clusterCSIDriver"]: 4,
+				},
+				cloudConfigOptions: map[string]int{
+					snapshotOptions["vvol"]["cloudConf"]: 4,
+				},
+				successfulSnapshotsCreated: 0,
+			},
+			{
+				name: "all snapshot options",
+				clusterCSIDriverOptions: map[string]int{
+					snapshotOptions["global"]["clusterCSIDriver"]: 5,
+					snapshotOptions["vsan"]["clusterCSIDriver"]:   10,
+					snapshotOptions["vvol"]["clusterCSIDriver"]:   15,
+				},
+				cloudConfigOptions: map[string]int{
+					snapshotOptions["global"]["cloudConf"]: 5,
+					snapshotOptions["vsan"]["cloudConf"]:   10,
+					snapshotOptions["vvol"]["cloudConf"]:   15,
+				},
+				successfulSnapshotsCreated: 0,
+			},
+		}
+
+		for _, t := range tests {
+			g.It(fmt.Sprintf("%s", t.name), func() {
+				for option, value := range t.clusterCSIDriverOptions {
+					e2e.Logf("updating %s to %d in clustercsidriver", option, value)
+					setClusterCSIDriverSnapshotOptions(oc, option, value)
+				}
+
+				for option, value := range t.cloudConfigOptions {
+					o.Eventually(func() error {
+						return loadAndCheckCloudConf(oc, "Snapshot", option, value)
+					}, time.Minute, time.Second).Should(o.Succeed())
+				}
+
+				if t.successfulSnapshotsCreated > 0 {
+					pvc, err := createTestPVC(oc, oc.Namespace(), "test-pvc", "1Gi")
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					_, err = createTestPod(oc, pvc.Name, oc.Namespace())
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					// Wait for pvc to be bound.
+					o.Eventually(func() error {
+						pvc, err := oc.AdminKubeClient().CoreV1().PersistentVolumeClaims(oc.Namespace()).Get(context.Background(), "test-pvc", metav1.GetOptions{})
+						if err != nil {
+							return err
+						}
+						if pvc.Status.Phase != v1.ClaimBound {
+							return fmt.Errorf("PVC not bound")
+						}
+						return nil
+					})
+
+					for i := 0; i < t.successfulSnapshotsCreated; i++ {
+						err := createSnapshot(oc, oc.Namespace(), fmt.Sprintf("test-snapshot-%d", i), "test-pvc")
+						o.Expect(err).NotTo(o.HaveOccurred())
+					}
+
+					// Next snapshot creation should be over the set limit and fail.
+					err = createSnapshot(oc, oc.Namespace(), "test-snapshot-failed", "test-pvc")
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					readyToUse, err := oc.Run("get").Args("volumesnapshot/test-snapshot-failed", "-o", "jsonpath={.status.readyToUse}").Output()
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					errMsg, err := oc.Run("get").Args("volumesnapshot/test-snapshot-failed", "-o", "jsonpath={.status.error.message}").Output()
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					e2e.Logf("VolumeSnapshot error message: %s readyToUse %s", errMsg, readyToUse)
+					if !strings.Contains(errMsg, "failed to take snapshot of the volume") && readyToUse != "false" {
+						e2e.Failf("VolumeSnapshot \"test-snapshot-failed\" should have failed and should not be ready to use")
+					}
+				}
```

Comment:
This should go to its own function, just like `loadAndCheckCloudConf`.

Reply:
Moved to separate function.